packages/grafana-runtime/src/utils/queryResponse.ts

```diff
@@ -73,10 +73,11 @@ export function toDataQueryResponse(
   }
 
   // If the response isn't in a correct shape we just ignore the data and pass empty DataQueryResponse.
-  if ((res as FetchResponse).data?.results) {
-    const results = (res as FetchResponse).data.results;
+  const fetchResponse = res as FetchResponse;
+  if (fetchResponse.data?.results) {
```

Comment:
This could be replaced by a guard to ensure type safety.

Reply:
you're right - i'll do that in a follow up PR if that's ok? since it will also affect a few more places 
![image](https://github.com/grafana/grafana/assets/20999846/ba01b075-b071-4c91-a317-66f5c8cb08c4)